espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/RouteMap.java

```diff
@@ -0,0 +1,39 @@
+package io.appium.espressoserver.lib.http;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import fi.iki.elonen.NanoHTTPD.Method;
+
+
+class RouteMap {
+
+    private final Map<Method, Map<String, RouteDefinition>> routeMap;
+
+    RouteMap() {
+        routeMap = new HashMap<>();
+
+    }
+
+    public void addRoute(RouteDefinition route) {
+        if (!routeMap.containsKey(route.getMethod())) {
+            routeMap.put(route.getMethod(), new HashMap<String, RouteDefinition>());
+        }
+        Map<String, RouteDefinition> methodMap = routeMap.get(route.getMethod());
+        methodMap.put(route.getRouteUri(), route);
+    }
+
+    public RouteDefinition findMatchingRoute(String uri) {
```

Comment:
java has a nice \@Nullable and \@NonNull annotations for such type of methods. Highly recommended ;)